samples/augment-reality-to-show-hidden-infrastructure/src/main/java/com/esri/arcgismaps/sample/augmentrealitytoshowhiddeninfrastructure/components/AugmentedRealityViewModel.kt

```diff
@@ -0,0 +1,162 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.augmentrealitytoshowhiddeninfrastructure.components
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.GeodeticCurveType
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.geometry.LinearUnit
+import com.arcgismaps.geometry.Polyline
+import com.arcgismaps.geometry.PolylineBuilder
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.ElevationSource
+import com.arcgismaps.mapping.NavigationConstraint
+import com.arcgismaps.mapping.symbology.MultilayerPolylineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.symbology.SolidStrokeSymbolLayer
+import com.arcgismaps.mapping.symbology.StrokeSymbolLayerLineStyle3D
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.SurfacePlacement
+import kotlinx.coroutines.launch
+
+class AugmentedRealityViewModel(app: Application) : AndroidViewModel(app) {
+
+    // Graphics overlay for the 3D pipes
+    val pipeGraphicsOverlay = GraphicsOverlay().apply {
+        sceneProperties.surfacePlacement = SurfacePlacement.Absolute
+    }
+
+    // Graphics overlay for the shadow of pipes underground
+    val pipeShadowGraphicsOverlay = GraphicsOverlay().apply {
+        opacity = 0.6f
+    }
+
+    // Graphics overlay for the leaders
+    val leaderGraphicsOverlay = GraphicsOverlay().apply {
+        sceneProperties.surfacePlacement = SurfacePlacement.Absolute
+    }
+
+    // Create a scene with an elevation source and grid and surface hidden
+    val arcGISScene = ArcGISScene().apply {
+        baseSurface.apply {
+            elevationSources.add(ElevationSource.fromTerrain3dService())
+            backgroundGrid.isVisible = false
+            opacity = 0.0f
+            navigationConstraint = NavigationConstraint.None
+        }
+    }
+
+    // Define a red 3D stroke symbol to show the pipe
+    private val pipeStrokeSymbol = SolidStrokeSymbolLayer(
+        width = 0.3,
+        color = Color.red,
+        lineStyle3D = StrokeSymbolLayerLineStyle3D.Tube
+    )
+    val pipeSymbol = MultilayerPolylineSymbol(listOf(pipeStrokeSymbol))
+
+    // Define a red 2D stroke symbol to show the pipe shadow
+    private val pipeShadowSymbol = SimpleLineSymbol(
+        style = SimpleLineSymbolStyle.Solid,
+        color = Color.red,
```

Comment:
Maybe helpful to differentiate the shadow with a different color? 
```suggestion
    // Define a yellow 2D stroke symbol to show the pipe shadow
    private val pipeShadowSymbol = SimpleLineSymbol(
        style = SimpleLineSymbolStyle.Solid,
        color = Color.yellow,
```